packages/promotion/src/models/promotion.ts

```diff
@@ -1,16 +1,71 @@
-import { generateEntityId } from "@medusajs/utils"
-import { BeforeCreate, Entity, PrimaryKey, OnInit } from "@mikro-orm/core"
+import { PromotionType } from "@medusajs/types"
+import { PromotionUtils, generateEntityId } from "@medusajs/utils"
+import {
+  BeforeCreate,
+  Entity,
+  Enum,
+  Index,
+  OnInit,
+  OneToOne,
+  OptionalProps,
+  PrimaryKey,
+  Property,
+  Unique,
+} from "@mikro-orm/core"
+import ApplicationMethod from "./application-method"
 
+type OptionalFields = "is_automatic"
+type OptionalRelations = "application_method"
 @Entity()
 export default class Promotion {
+  [OptionalProps]?: OptionalFields | OptionalRelations
+
   @PrimaryKey({ columnType: "text" })
   id!: string
 
+  @Property({ columnType: "text" })
+  @Unique({
+    name: "IDX_promotion_code_unique",
+    properties: ["code"],
```

Comment:
suggestion: add index as well, constraint and index does not act exactly the same